### ISSUE TITLE
Stuck in PumpThread() in .Net MF 4.3

### ISSUE THREAD

**Issue by michael-chi** [Azure/amqpnetlite#155]
I am using AMQP.Net Lite to connect to Azure IOT Hub in .Net Micro Framework 4.3 & 4.2.
In my environment (VS 2015 Enterprise/Windows 10 Build 14931/Windows Server 2012 R2 on Azure/.Net Micro Framework 4.4/4.3/4.2) I found I need to add a Thread.Sleep(0) in Connection.cs\Pump class in order to have my program works. Otherwise it stuck in the while loop.

so the PumpThread() method in Connection.cs would look like below
//...omitted...
                while (this.connection.state != State.End)
                {
                    Thread.Sleep(0);//Michael
                    try
                    {
                        ByteBuffer buffer = Reader.ReadFrameBuffer(this.connection.transport, sizeBuffer, this.connection.maxFrameSize);
                        this.connection.OnFrame(buffer);
                    }
                    catch (Exception exception)
                    {
                        this.connection.OnIoException(exception);
                    }
                }

Below is a code snap from my test project, it's a simple sample codes on the internet
=== My Main() ===
        static void Main(string[] args)
        {
            Amqp.Trace.TraceLevel = Amqp.TraceLevel.Frame | Amqp.TraceLevel.Verbose;
#if NETMF
            Amqp.Trace.TraceListener = (f, a) => Debug.Print(DateTime.Now.ToString("[hh:ss.fff]") + " " + Fx.Format(f, a));
#else
            Amqp.Trace.TraceListener = (f, a) => System.Diagnostics.Trace.WriteLine(DateTime.Now.ToString("[hh:ss.fff]") + " " + Fx.Format(f, a));
#endif
            string audience = Fx.Format("{0}/devices/{1}", HOST, DEVICE_ID);
            string resourceUri = Fx.Format("{0}/devices/{1}", HOST, DEVICE_ID);
            string sasToken = SAS_TOKEN;
#if false
            address = new Address(HOST, PORT, "device003@sas.winstron-demo-01",
            SAS_TOKEN,
            "AMQPS");
            connection = new Connection(address);
            bool cbs = true;
#else
            address = new Address(HOST, PORT, null, null);
            connection = new Connection(address);
            bool cbs = PutCbsToken(connection, HOST, sasToken, audience);
#endif
            if (cbs)
            {
                session = new Session(connection);
                SendEvent();
                receiverThread = new Thread(ReceiveCommands);
                receiverThread.Start();
            }
            // just as example ...
            // the application ends only after received a command or timeout on receiving
            receiverThread.Join();
            session.Close();
            connection.Close();
        }

**Comment by xinchen10** [comment]
@michael-chi  are you running the test on a device or in the emulator?

**Comment by michael-chi** [comment]
running in the emulator

**Comment by xinchen10** [comment]
The issue is described here: http://amqpnetlite.codeplex.com/wikipage?title=NETMF%20and%20TLS%2fSSL%20connection&referringTitle=Documentation

The fix I proposed was actually taken by the netmf team.
https://github.com/NETMF/netmf-interpreter/issues/330

But the fix is in the dev branch, not master. I think the 4.4 SDK was released from master so none of the SDKs have the fix.

I don't know why adding a sleep helps but it is a nice trick to make it run on the emulator.
